### PR TITLE
Document how we handle stale tickets and PRs

### PR DESCRIPTION
**Pull request by ChristophWurst**
This is a proposal for both the process and the specific configurations.

Todo
* [ ] Find consent on what we want to automate
* [ ] Find consent on the timeouts
* [x] Improve the comments so they link back to the process

Fixes https://github.com/nextcloud/groupware/issues/9

### CODE REVIEW

File: processes/stale_issues_and_prs.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+In the rare case that a ticket is not updated by anyone we will close it after a long grace period. This still doesn't mean the ticket can't be reconsidered. Maintainers can reopen at any time and relabel accordingly.
+
+## Automation
+
+* Runs daily at 3:00AM
+* Tickets
+  * Marked as stale if
+    1. Hasn't received any update in three weeks
+    2. Does not have any status label (to develop, developing, review) assigned. So this is either still to triage or doesn't have any tags at all
+  * A comment is left for the main repo owner to please triage the ticket
```
Comment (GretaD) [inline processes/stale_issues_and_prs.md:20]: when does this step happen?

Comment (ChristophWurst) [reply processes/stale_issues_and_prs.md]: I hope it's clearer now

File: processes/stale_issues_and_prs.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+* Runs daily
+* Tickets
+  * Stale after three weeks
+    * Tickets with a status label (to develop, developing, review) are not considered stale. It's only unlabeled ones and the ones to triage
+    * A comment is left for the main repo maintainer to please triage the ticket
+  * Closed after five weeks of no activity
+* Pull requests
+  * Stale after a week
+    * A comment is left for the mani repo maintainer to please have a look
```
Comment (miaulalala) [inline processes/stale_issues_and_prs.md:23]:
```suggestion
    * A comment is left for the main repo maintainer to please have a look
```

File: processes/stale_issues_and_prs.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v4
+        with:
+          # Issues
+          days-before-issue-stale: 21
+          days-before-issue-close: 35
+          stale-issue-label: 'stale'
+          stale-issue-message: 'This ticket had no activity recently. @ChristophWurst please adjust the labels if this ticket should stay open and close it otherwise. [Learn more about the stale process](https://github.com/nextcloud/groupware/blob/main/processes/stale_issues_and_prs.md)'
+          close-issue-message: 'This ticket had no activity for a while, so we close it. If there was information missing on a bug report, we can re-open once the info is provided. [Learn more about the stale process](https://github.com/nextcloud/groupware/blob/main/processes/stale_issues_and_prs.md)'
```
Comment (miaulalala) [inline processes/stale_issues_and_prs.md:44]:
```suggestion
          close-issue-message: 'This ticket has had no activity for a while, so we are closing it. If this is due to information missing on a bug report, we can re-open the ticket once the info is provided. [Learn more about the stale process](https://github.com/nextcloud/groupware/blob/main/processes/stale_issues_and_prs.md)'
```

**Review by miaulalala — APPROVED** [review]
Looks good to me. But are you happy handling being the only one tagged @ChristophWurst ?

**Comment by ChristophWurst** [comment]
> Looks good to me. But are you happy handling being the only one tagged @ChristophWurst ?

It's the main repo owner. I will only get pinged for Contacts.

**Review by GretaD — APPROVED** [review]

File: processes/stale_issues_and_prs.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          close-issue-message: 'This ticket has had no activity for a while, so we are closing it. If this is due to information missing on a bug report, we can re-open the ticket once the info is provided. [Learn more about the stale process](https://github.com/nextcloud/groupware/blob/main/processes/stale_issues_and_prs.md)'
+          exempt-issue-labels: '1. to develop,2. developing,3. to review,4. to release'
+          # PRs
+          days-before-pr-stale: 7
+          days-before-pr-close: 180
+          stale-pr-label: 'stale'
+          stale-pr-message: 'This pull request needs attention. @ChristophWurst have a look please! [Learn more about the stale process](https://github.com/nextcloud/groupware/blob/main/processes/stale_issues_and_prs.md)'
+          close-pr-message: 'This pull request has had no activity for a while, so we are closing it. If the changes are still applicable and should be integrated, please get in contact with the maintainers, they can reopen. [Learn more about the stale process](https://github.com/nextcloud/groupware/blob/main/processes/stale_issues_and_prs.md)'
```
Comment (st3iny) [inline processes/stale_issues_and_prs.md:51]:
```suggestion
          close-pr-message: 'This pull request has had no activity for a while, so we are closing it. If the changes are still applicable and should be integrated, please get in contact with the maintainers, so they can reopen. [Learn more about the stale process](https://github.com/nextcloud/groupware/blob/main/processes/stale_issues_and_prs.md)'
```

**Review by st3iny — APPROVED** [review]

**Comment by st3iny** [comment]
> This pull request needs attention. @ChristophWurst have a look please! [Learn more about the stale process](https://github.com/nextcloud/groupware/blob/main/processes/stale_issues_and_prs.md)

Oh the irony ...

**Pull request merged**